src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -169,25 +170,33 @@ class ErgoWalletActor(ergoSettings: ErgoSettings) extends Actor with ScorexLoggi
           R5 -> StringConstant(description),
           R6 -> IntConstant(decimals)
         )
-        new ErgoBoxCandidate(0L, address.script, Seq(assetId -> amount), nonMandatoryRegisters)
+        val lockWithAddress = (addressOpt orElse trackedAddresses.headOption)
+          .getOrElse(throw new Exception("No address available for box locking"))
+        val candidateMock =
+          new ErgoBoxCandidate(100000L, lockWithAddress.script, Seq(assetId -> amount), nonMandatoryRegisters)
+        val minimalErgoAmount = minimalValueOf(candidateMock)
+        new ErgoBoxCandidate(minimalErgoAmount, candidateMock.proposition,
+          candidateMock.additionalTokens, candidateMock.additionalRegisters)
       case other => throw new Exception(s"Unknown TransactionRequest type: $other")
     }
   }
 
-  protected def generateTransactionWithOutputs(requests: Seq[TransactionRequest]): Try[ErgoTransaction] =
+  def generateTransactionWithOutputs(requests: Seq[TransactionRequest]): Try[ErgoTransaction] =
     requestsToBoxCandidates(requests).map { payTo =>
       require(prover.dlogPubkeys.nonEmpty, "No public keys in the prover to extract change address from")
       require(requests.count(_.isInstanceOf[AssetIssueRequest]) <= 1, "Too many asset issue requests")
-      require(payTo.forall(_.value >= 0), "Non-positive Ergo value")
-      require(payTo.forall(_.additionalTokens.forall(_._2 >= 0)), "Non-positive asset value")
+      require(payTo.forall(c => c.value >= minimalValueOf(c)), "Minimal ERG value not met")
+      require(payTo.forall(_.additionalTokens.forall(_._2 >= 0)), "Negative asset value")
 
       val assetIssueBox = payTo
         .zip(requests)
         .filter(_._2.isInstanceOf[AssetIssueRequest])
         .map(_._1)
         .headOption
 
-      val targetBalance = payTo.map(_.value).sum
```

Comment:
here one-line syntax looks quite clear, why to change it?

Reply:
To make this block of code look consistently